dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp

```diff
@@ -1161,7 +1162,10 @@ void DAGStorageInterpreter::buildLocalExec(
             disaggregated_snap->addTask(table_id, std::move(table_snap));
         }
 
-        group_builder.merge(std::move(builder));
+        if (has_multiple_partitions)
+            builder_pool.add(builder);
+        else
+            group_builder.merge(std::move(builder));
```

Comment:
Is this `group_builder.merge(std::move(builder)` still needed? In my understanding, if has_multiple_partitions is false, we can build `buildLocalExecForPhysicalTable` with `group_builder` directly?

Reply:
This is just to deal with a case: the number of streams allocated to one index is 1, although there is no such situation at present...

Reply:
It’s really not necessary, just follow the original code here.
And the code of streams as follows
```
        if (has_multiple_partitions)
            stream_pool->addPartitionStreams(current_pipeline.streams);
        else
            pipeline.streams.insert(pipeline.streams.end(), current_pipeline.streams.begin(), current_pipeline.streams.end());
```